torch/testing/_internal/common_methods_invocations.py

```diff
@@ -303,6 +309,52 @@ def sample_inputs(self, device, dtype, requires_grad=False, **kwargs):
             samples = self.sample_inputs_func(self, device, dtype, requires_grad)
         return samples
 
+    def sample_inputs(self, device, dtype, requires_grad=False, **kwargs):
+        """Generates a sequence of SampleInputs.
+
+        The OpInfo keeps track the the sample which is currently yielded: wrap
+        your code inside a with op.decorate_errors() to get better exception
+        messages which include the current sample.
+
+        If subclasses want to customize the generation of inputs, they should
+        override generate_sample_inputs(). On the contrary, end users should
+        always call sample_inputs().
+        """
+        if self._current_sample is not None:
+            raise ValueError('Cannot call op.sample_inputs() while another '
+                             'iteration is in progress')
+        samples = self.generate_sample_inputs(device, dtype, requires_grad, **kwargs)
+        for i, sample in enumerate(samples):
+            assert isinstance(sample, SampleInput)
+            # make sure that every SampleInput has a sensible name: if the
+            # name was not given, use the index instead
+            if sample.name is None:
+                sample.name = str(i)
+            self._current_sample = sample
+            yield sample
+        self._current_sample = None
+
+    def get_one_sample_input(self, device, dtype, requires_grad=False, **kwargs):
+        """Returns a single SampleInput if it exists, else None"""
+        samples = self.generate_sample_inputs(device, dtype, requires_grad, **kwargs)
+        if len(samples) == 0:
+            return None
+        return samples[0]
+
+    @contextlib.contextmanager
+    def decorate_errors(self):
+        """Attach additional information to exceptions.
+
+        In particular, display also the name of the current SampleInput if the
+        code is in the middle of a loop "for sample in op.sample_inputs():"
+        """
+        try:
+            yield
+        except Exception as exc:
+            if self._current_sample:
+                exc.args = (f'[SampleInput: {self._current_sample.name}] {exc}',)
```

Comment:
With @kshitij12345's PR I think this can just print the sample input's summary

Reply:
Found it! This one: https://github.com/pytorch/pytorch/pull/58295

I'll try and merge it soon

Reply:
see https://github.com/pytorch/pytorch/pull/58481#issuecomment-850537695 for a detailed answer